### PR TITLE
React Aliases

### PR DESCRIPTION
**Pull request by wyattjoh**
This adds aliases to the new react packages being used by Next.js.

### CODE REVIEW

**Comment by wyattjoh** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #57161** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57161" target="_blank">Graphite</a> 
    * **PR #57162** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57162" target="_blank">Graphite</a> 
      * **PR #57283** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57283" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/57283?utm_source=stack-comment).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 3d1fe2d89cf9f4d577df1eb6131afa1587a4f82a

`pnpm test test/production/pages-dir/production/test/index.test.ts` 
- Production Usage > should output traces

<details>
<summary>Expand output</summary>

  ● Production Usage › should output traces

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      134 |           file.includes('next/dist/server/send-payload/index.js')
      135 |         )
    > 136 |       ).toBe(true)
          |         ^
      137 |       expect(
      138 |         serverTrace.files.some((file) =>
      139 |           file.includes('next/dist/server/lib/route-resolver.js')

      at Object.toBe (production/pages-dir/production/test/index.test.ts:136:9)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| buildDuration | 10.5s | 10.7s | ⚠️ +193ms |
| buildDurationCached | 6.2s | 6.3s | ⚠️ +102ms |
| nodeModulesSize | 174 MB | 178 MB | ⚠️ +3.02 MB |
| nextStartRea..uration (ms) | 496ms | 499ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.9 kB | 27.9 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.2 kB | 53.2 kB | N/A |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.5 kB | 45.5 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 33 kB | 33 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 45.7 kB | 45.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.59 kB | 2.59 kB | ✓ |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.38 kB | 4.38 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 316 B | 318 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.99 kB | 3.99 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| index.html gzip | 528 B | 529 B | N/A |
| link.html gzip | 541 B | 541 B | ✓ |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 541 B | 541 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.7 kB | 95.3 kB | N/A |
| page.js gzip | 158 kB | 158 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/react-bump | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 624 B | ✓ |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.9 kB | 22.9 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.55 kB | 2.55 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 3d1fe2d89cf9f4d577df1eb6131afa1587a4f82a

**Review by ztanner — APPROVED** [review]

File: packages/next/taskfile-swc.js [hunk 1]
```diff
@@ -100,6 +100,15 @@ module.exports = function (task) {
...
           },
+          baseUrl: __dirname,
+          paths: {
+            'react-dom/server.edge': [
+              './compiled/react-dom-experimental/server.edge',
+            ],
+            'react-dom/static.edge': [
+              './compiled/react-dom-experimental/static.edge',
+            ],
+          },
```
Comment (gnoff) [inline packages/next/taskfile-swc.js:111]: I've not seen us do this kind of aliasing for react before. Why is this required for this PR?

**Review by gnoff — COMMENTED** [review]
I noticed our swc taskfile is targeting node 16 still. Aren't we bumping minimum node version to 18 with next 14? maybe we should bump this too?
